### ISSUE TITLE
ExpressionBodySyntax discards explicit return type

### ISSUE THREAD

**Issue by seanf** [arturbosch/detekt#87]
This function:
```kotlin
    fun retrieveDefaultNativeName(): String {
        return asULocale().getDisplayName(asULocale())
    }
```
becomes this after ExpressionBodySyntax formats it:
```kotlin
    fun retrieveDefaultNativeName() = asULocale().getDisplayName(asULocale())
```
This change leads to the inspection warning in IntelliJ: "Declaration has type inferred from a platform call, which can lead to unchecked nullability issues. Specify type explicitly as nullable or non-nullable."

_This was with detekt M11._

**Comment by arturbosch** [comment]
Good point maybe noch deleting the return type would be the fix. Or just deleting them in member functions bit not at top level?

**Comment by seanf** [comment]
I've seen it recommended to specify the type always because it's part of
the API (unless it's a private function, I suppose).

And, private or not, there's always a risk of introducing that inspection
warning if the inferred type is derived from a platform type.

So I think it would be better not to delete the return type at all. I think
I would find it pretty surprising to have detekt delete my explicit type
information in any case.
